src/components/capture/creation/SpecForm.tsx

```diff
@@ -78,28 +83,32 @@ function NewCaptureSpecForm(props: NewCaptureSpecFormProps) {
                 {error}
             </Alert>
         );
-    } else if (specSchema.type) {
-        const uiSchema = generateCustomUISchema(specSchema);
+    } else if (endpointSchema.type) {
+        const uiSchema = generateCustomUISchema(endpointSchema);
+        const showValidationVal = showValidation(displayValidation);
+        const handlers = {
+            onChange: (form: any) => {
+                if (!isEmpty(form.data)) {
+                    dispatch({
+                        payload: form,
+                        type: ActionType.CAPTURE_SPEC_CHANGED,
+                    });
+                }
+            },
+        };
```

Comment:
Two of these were inline with the JSX... think it makes it a bit cleaner. Handlers are only needed when the form is going to display so that is why they are currently down here and not up above.